src/core/model/reducer.ts

```diff
@@ -45,6 +46,21 @@ export const uiSchemaReducer = (
   switch (action.type) {
     case SET_UISCHEMA:
       return buildLinkedUiSchemaTree(action.uiSchema);
+    case ADD_UI_SCHEMA_ELEMENT_TO_LAYOUT:
+      const newUiSchema = cloneTree(action.layout as LinkedUISchemaElement);
+      if (isPathError(newUiSchema)) {
+        console.error(
+          'An error occured when cloning the ui schema',
+          newUiSchema
+        );
+        // Do nothing
+        return action.layout;
+      }
+      const newUIElement = action.uiSchemaElement;
```

Comment:
Here is an inconsistency to the current approach: In `ADD_SCHEMA_ELEMENT_TO_LAYOUT` we create the control within the reducer, while here the ui schema element is already created in the drag operation. We should probably make that consistent, however I have no hard opinion on where to do that. Probably always already in the drag operation? What do you guys think?

Reply:
I would also prefer to do it in the drag operation.

Reply:
by creating the control in the drag operation we move logic from the reducer to the ui
If we do so, then the reducer should be unified as in both cases we add a 'LinkedUISchemaElement' to the layout. the control needs additional handling but that's it.

Reply:
Hmm still it's not the same as one action only has an effect on the ui schema and the other one on both schema and ui schema (at least in our models, not the exported versions).

You are right that we would be moving logic into the ui which is not that great. Therefore I would prefer creating the element in the reducer for now.

Reply:
hmm, maybe we should rename the actions then as the new action is called: `ADD_UI_SCHEMA_ELEMENT_TO_LAYOUT` and this would include the control case covered by `ADD_SCHEMA_ELEMENT_TO_LAYOUT`